### PR TITLE
Add option for adding OMSCloudID and AzureResourceID

### PR DESCRIPTION
**Pull request by sugr4**
to http headers on onboarding

@Microsoft/omsagent-devs

### CODE REVIEW

File: installer/scripts/omsadmin.sh [hunk 1]
```diff
@@ -416,8 +422,40 @@ onboard()
 
     set_proxy_setting
 
-    # Send the request to the registration server
-    RET_CODE=`curl --header "x-ms-Date: $REQ_DATE" \
+    if [ -n "$AZURE_RESOURCE_ID" ]; then
```
Comment (robbiezhang) [inline installer/scripts/omsadmin.sh]: you don't need to add them into this header

File: installer/scripts/omsadmin.sh [hunk 2]
```diff
@@ -416,8 +422,40 @@ onboard()
...
+        if [ -f /sys/devices/virtual/dmi/id/chassis_asset_tag ]; then
+            ASSET_TAG=$(cat /sys/devices/virtual/dmi/id/chassis_asset_tag)
+            case "$ASSET_TAG" in
+                77*) OMSCLOUD_GUID=$ASSET_TAG ;;       # If the asset tag begins with a 77 this the azure guid
```
Comment (robbiezhang) [inline installer/scripts/omsadmin.sh]: always get the GUID and store it into the omsadmin.conf no matter the AzureResourceId is specified or not.

File: installer/scripts/omsadmin.sh [hunk 3]
```diff
@@ -116,6 +118,7 @@ save_config()
     echo URL_TLD=$URL_TLD >> $CONF_OMSADMIN
     echo DSC_ENDPOINT=$DSC_ENDPOINT >> $CONF_OMSADMIN
     echo OMS_ENDPOINT=https://$WORKSPACE_ID.ods.$URL_TLD/OperationalData.svc/PostJsonDataItems >> $CONF_OMSADMIN
+    echo AZURE_RESOURCE_ID=$AZURE_RESOURCE_ID >> $CONF_OMSADMIN
```
Comment (robbiezhang) [inline installer/scripts/omsadmin.sh]: also save the OMSCloudID

File: installer/scripts/omsadmin.sh (hunk 1 above)
Comment (robbiezhang) [inline installer/scripts/omsadmin.sh]: can you also get the UUID here?

**Comment by sugr4** [comment]
Made changes based on comments

**Comment by robbiezhang** [comment]
Signoff

File: installer/bundle/bundle_skel.sh [hunk 4]
```diff
@@ -558,6 +559,11 @@ do
             shift 1
             ;;
 
+        -a|--id)
```
Comment (varunkumta) [inline installer/bundle/bundle_skel.sh]: --id is also used for the workspace id -w|--id so we should probably use another string here

Comment (robbiezhang) [inline installer/bundle/bundle_skel.sh]: good catch.

**Pull request merged**